src/openstack_cloud/openstack_manager.py

```diff
@@ -532,20 +532,33 @@ def _ssh_health_check(conn: OpenstackConnection, server_name: str, startup: bool
             logger.error("[ALERT]: Unable to SSH to server: %s, reason: %s", server_name, str(exc))
             raise
 
-        result: invoke.runners.Result = ssh_conn.run("ps aux", warn=True)
-        logger.debug("Output of `ps aux` on %s stderr: %s", server_name, result.stderr)
-        if not result.ok:
+        cloud_init_result: invoke.runners.Result = ssh_conn.run("cloud-init status", warn=True)
+        logger.debug(
+            "Output of `cloud-init status` on %s stderr: %s", server_name, cloud_init_result.stderr
+        )
+        if not cloud_init_result.ok:
+            logger.warning("cloud-init status command failed on %s.", server_name)
+            raise _SSHError(f"cloud-init status command failed on {server_name}.")
+        # cloud-init status should be "running"
```

Comment:
Based on this comment I'd expect the line below to be `if "running" not in cloud_init_result.stdout`. What do you think?